src/org/opensolaris/opengrok/index/Indexer.java

```diff
@@ -371,8 +387,13 @@ public static WebAddress parseWebAddress(String webAddr) {
                 String.format("\nUsage: java -jar %s [options] [subDir1 [...]]\n", program));
 
             parser.on("-?", "-h", "--help", "Display this usage summary.").Do( v -> {
-                parser.help();
-                System.exit(status);
+                help = true;
+                helpUsage = parser.getUsage();
+            });
+
+            parser.on("--detailed",
+                "Display additional help with -h,--help.").Do(v -> {
+                helpDetailed = true;
```

Comment:
What should happen if `--detailed` is used without `-h` ? Trigger an error or imply `help = 1` ? (in the latter case `--detailed-help` might be better)

Reply:
It would be ignored, which I thought was OK.

Reply:
I think we can do better than that, especially if it costs one line of code.

Reply:
Hmm. It's documented as "additional help with -h,--help". Why raise an exception if `-h,--help` is not used?

To compare with another option, `-V,--version` just "Print[s] version and quit" — but there is no exhaustive checking that `-V,--version` is not used with any other options. Likewise with `--man` to "Generate OpenGrok XML manual page [and quit]".

Reply:
I was actually advocating for the reverse - pretend that `-h` is set if `--detailed` stands alone. The current option parsing framework does not give the flexibility to state incompatible options so feel free to leave as is.

Reply:
OK